Sources/PT.PM.Matching/Patterns/PatternReturn.cs

```diff
@@ -0,0 +1,47 @@
+﻿using PT.PM.Common;
+using PT.PM.Common.Nodes;
+using PT.PM.Common.Nodes.Statements;
+using System;
+
+namespace PT.PM.Matching.Patterns
+{
+    public class PatternReturn : PatternUst
+    {
+        public PatternUst Pattern { get; set; } = PatternAny.Instance;
+
+        public PatternReturn()
+        {
+        }
+
+        public PatternReturn(PatternUst pattern, TextSpan textSpan = default(TextSpan))
+            : base(textSpan)
+        {
+            Pattern = pattern ?? throw new ArgumentNullException(nameof(pattern));
+        }
+
+        public override string ToString() => $"return {Pattern}";
+
+        public override MatchingContext Match(Ust ust, MatchingContext context)
+        {
+            if (ust == null)
+            {
+                return context.Fail();
+            }
+
+            if (ust is ReturnStatement returnStatement)
+            {
+                MatchingContext newContext = Pattern.Match(returnStatement.Return, context);
+                if (newContext.Success)
+                {
+                    return newContext.AddMatch(ust);
+                }
+                else
+                {
+                    return newContext.Fail();
+                }
+            }
```

Comment:
Use the following common style (see also other pattern nodes):

```
if (ust is ReturnStatement returnStatement)
{
    ...
}
else
{
    return newContext.Fail();
}
```

instead of early null check and break.